### PR TITLE
feat(IDX): bazel coverage in PR run / upload to S3

### PR DESCRIPTION
**Pull request by marko-k0**
Allow running `bazel coverage` on PR when `CI_COVERAGE` label is set. Report is uploaded to S3 and URL logged and set as run annotation. 

image

### CODE REVIEW

File: ci/scripts/bazel-coverage.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+set -eEuo pipefail
+
+bazel query --universe_scope=//... \
```
Comment (marko-k0) [inline ci/scripts/bazel-coverage.sh:4]: I'll look whether there's any change with having functional rust code coverage with wasm32 target in the follow up.

File: bazel/conf/.bazelrc.build [hunk 2]
```diff
@@ -22,6 +22,7 @@ build --@rules_rust//:rustfmt.toml=//:rustfmt.toml
 # lockfile for MODULE.bazel
 common --lockfile_mode=off
 
+
```
Comment (basvandijk) [inline bazel/conf/.bazelrc.build:25]: accident?

**Review by basvandijk — APPROVED** [review]
LGTM.

I was just wondering why not use [GitHub workflow artifacts](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/storing-and-sharing-data-from-a-workflow) for storing coverage reports? That would seem like a simpler solution since it won't require S3.

Is the default retention of 90 days the issue here?

File: bazel/conf/.bazelrc.build (hunk 2 above)
Comment (marko-k0) [reply bazel/conf/.bazelrc.build:25]: Yes.

**Comment by marko-k0** [comment]
> LGTM.
> 
> I was just wondering why not use [GitHub workflow artifacts](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/storing-and-sharing-data-from-a-workflow) for storing coverage reports? That would seem like a simpler solution since it won't require S3.
> 
> Is the default retention of 90 days the issue here?

Yes, we'd need to change 90 days retention if we want to retain reports for longer.

**Pull request merged**